### ISSUE TITLE
Automatic retries beyond `read_parquet` / `to_parquet`

### ISSUE THREAD

**Issue by jrbourbeau** [dask/dask#9594]
In https://github.com/dask/dask/pull/9175 we added automatic retries when using remote filesystems with `dd.read_parquet` / `dd.to_parquet`. It'd be nice for other I/O functions (e.g. `dd.read_csv` / `dd.to_csv`) to have similar robustness.

**Comment by gjoseph92** [comment]
One thing to be aware of is https://github.com/dask/dask/issues/7036. For DataFrames, currently fusion is off by default https://github.com/dask/dask/pull/7620, but there are reasons why we might want to turn it back on (https://github.com/dask/dask/issues/8447). And if anyone loads data in a DataFrame, then turns it into an Array/Bag down the line, then they'll silently lose the retries.

**Comment by jrbourbeau** [comment]
Totally agree there are some known issues. I still think it's worth adding auto-retries, we just know there are some known issues. Worst case scenario, users will have the current behavior on `main` (i.e. retries will be lost)

**Comment by gjoseph92** [comment]
Agreed. It would probably just be good to 1) document the auto-retries and 2) mention this caveat in those docs.

**Comment by ncclementi** [comment]
When trying to implement retries over at #9601 the initial approach was to annotate read_bytes around here https://github.com/dask/dask/blob/e5b27eb2037e79ff31b935ba89976ab52b946605/dask/bytes/core.py#L145-L159

those annotations were not carried over, in other words when running a test that look like this 

```python
    def test_read_csv_retries(tmpdir):

    fn = str(tmpdir)
    remote_fn = f"simplecache://{tmpdir}"
    storage_options = {"target_protocol": "file"}
    
    ddf2 = dd.read_csv(f"{remote_fn}/*.part", storage_options=storage_options)
    layer = hlg_layer(ddf2.dask, "read-csv")
    assert layer.annotations
    assert layer.annotations["retries"] == 5
```

There were no annotations. We know that the annotations were created on read_bytes because we checked with breakpoints at that level. 

Found a workaround, see PR which requires Adding storage options to text_blocks_to_pandas is breaking other tests because I'm not passing it into those tests, I want to make sure there is no way around this before I start touching all the other tests. It seems this way it'll be just a bandaid. Thoughts? 

cc: @mrocklin

**Comment by ncclementi** [comment]
https://github.com/dask/dask/pull/9601#issuecomment-1304141674 for visibity
